src/charm.py

```diff
@@ -105,12 +111,18 @@ def _on_update_status(self, _):
             self.unit.status = BlockedStatus("Missing relation to config-server.")
             return
 
-            # restart on high loaded databases can be very slow (e.g. up to 10-20 minutes).
+        if self.cluster.get_tls_statuses():
+            self.unit.status = self.cluster.get_tls_statuses()
+            return
+
+        # restart on high loaded databases can be very slow (e.g. up to 10-20 minutes).
         if not self.cluster.is_mongos_running():
             logger.info("mongos has not started yet")
             self.unit.status = WaitingStatus("Waiting for mongos to start.")
             return
 
+        self.unit.status = ActiveStatus()
```

Comment:
this part I am uncertain about - I know we want to avoid clearing statuses. 

But shouldn't the update status hook should check all status and report active if the service is active?

Reply:
Ideally we'd have a full reconcile method that go through all checks and clear / keep accordingly.
However, Correct me if I'm wrong, but I believe that given that most hooks which set a temporary status (waiting, maintenance, blocked) end up deferring to resolve the temporary state, they already take care of the clearing of the statuses. 
So i think that on update status we should just clear whatever statuses update_status might have set

Reply:
Unfortunately:
> However, Correct me if I'm wrong, but I believe that given that most hooks which set a temporary status (waiting, maintenance, blocked) end up deferring to resolve the temporary state, they already take care of the clearing of the statuses.

Many hooks in the charm do not clear their own statuses, most charms in the MongoDB suite rely on update-status to clear their statuses. 

I am open to adopting the approach you suggest but it is a bigger change

Reply:
Okay, let's leave it to another PR